core/arch/arm/mm/core_mmu.c

```diff
@@ -959,6 +959,10 @@ static void check_pa_matches_va(void *va, paddr_t pa)
 		return;
 	}
 #ifdef CFG_WITH_PAGER
+	if (v >= CFG_TEE_LOAD_ADDR && v < core_mmu_linear_map_end) {
+		TEE_ASSERT(v == pa);
```

Comment:
agree. discard my comment.
i note to have a look a assert() vs TEE_ASSERT(). they are redundant but regarding verbosity.